File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        int left = 0; int right = s.length() - 1;
        s = s.toLowerCase();
        while(left < right){
            char l = s.charAt(left);
            char r = s.charAt(right);
            if(!Character.isLetterOrDigit(l)) left++;
            else if(!Character.isLetterOrDigit(r)) right--;
            else{
                if(l != r) return false;
                left++;
                right--;
            }
        }
        return true;
    }
    
    public boolean check(char l, char r){
        return Math.abs(l - r) == 0 || Math.abs(l - r) == 32;
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = s.lower()
        left = 0
        right = len(s) - 1
        while left < right:
            c = s[left]
            d = s[right]
            if not c.isalnum():
                left += 1
            elif not d.isalnum():
                right -= 1
            elif c == d:
                left += 1
                right -= 1
            else:
                return False
        return True